libkbfs/kbpki_client.go

```diff
@@ -216,8 +217,36 @@ func (k *KBPKIClient) GetCurrentMerkleSeqNo(ctx context.Context) (
 
 // IsTeamWriter implements the KBPKI interface for KBPKIClient.
 func (k *KBPKIClient) IsTeamWriter(
-	ctx context.Context, tid keybase1.TeamID, uid keybase1.UID) (bool, error) {
-	teamInfo, err := k.serviceOwner.KeybaseService().LoadTeamPlusKeys(ctx, tid)
+	ctx context.Context, tid keybase1.TeamID, uid keybase1.UID,
+	verifyingKey kbfscrypto.VerifyingKey) (bool, error) {
+	// Use the verifying key to find out the eldest seqno of the user.
+	userInfo, err := k.loadUserPlusKeys(ctx, uid, verifyingKey.KID())
+	if err != nil {
+		return false, err
+	}
+
+	found := false
+	for _, key := range userInfo.VerifyingKeys {
```

Comment:
It could be, but that's kind of a large-ish change within the package.  Do you think it's worth it?